lib/pilot/build_manager.rb

```diff
@@ -58,45 +58,69 @@ def describe_build(build)
       return row
     end
 
-    # This method will takes care of checking for the processing builds every few seconds
-    # @return [Build] The build that we just uploaded
-    def wait_for_processing_build
-      # the upload date of the new buid
+    def obtain_uploaded_build
+      # the upload date of the new build
       # we use it to identify the build
+      # or use config[:build_full_version] if provided
 
       start = Time.now
-      wait_processing_interval = config[:wait_processing_interval].to_i
-      latest_build = nil
-      loop do
-        Helper.log.info "Waiting for iTunes Connect to process the new build"
-        sleep wait_processing_interval
-        builds = app.all_processing_builds
-        break if builds.count == 0
-        latest_build = builds.last # store the latest pre-processing build here
+
+      processing_build = wait_for_processing_build
+      full_build = wait_for_full_build(processing_build)
+
+      if full_build
+        minutes = ((Time.now - start) / 60).round
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.